flink-clients/src/main/java/org/apache/flink/client/cli/CliFrontendParser.java

```diff
@@ -371,6 +372,18 @@ public static void printHelpForRun(Collection<CustomCommandLine> customCommandLi
 		System.out.println();
 	}
 
+	public static void printHelpForRunApplication() {
+		HelpFormatter formatter = new HelpFormatter();
+		formatter.setLeftPadding(5);
+		formatter.setWidth(80);
+
+		System.out.println("\nAction \"run-application\" runs an application in Application Mode.");
+		System.out.println("\n  Syntax: run-application -t [kubernetes-application, yarn-application] [OPTIONS] <jar-file> <arguments>");
```

Comment:
I think these would come from the description of the `-t` parameter, right?

I also noticed that we're hardcoding the `foo-application` targets/executors because they are not real executors. That doesn't seem right.

Maybe it was a mistake how we separated job/session/application. "Job" and "session" are executors, while "application" is a whole different mode. 🙈

Reply:
For the `-t` I put them here because these are the only valid options for `run-application`. The rest are not compatible with the `run-application` command.

The reason for hardcoding the `foo-application` is a combination of what you said, and also the package structure of the project that makes it impossible to find all available executors without service discovery.

Now for the separation, you mean that we should not have a separate command (like `run-application`) or something else?

Reply:
No, unfortunately I think we need the `run-application` because it can't be implemented as "just an executor".

Why wouldn't it work with service discovery? I know that right now there is nothing that we could load that could report the name but we could add sth? I just doesn't feel very extensible to have the names hardcoded here. 😅

Reply:
For the `run-application` I agree, and this would be my answer actually if you suggested to use the `run` for everything :)

I agree also about the extensibility for the general help message of the `-t`. I may need to think though how to add it. For the `run-application` though, we should still hardcode it because we want to say that the user can only use a subset of the available executors with this command. WDYT?